apiary.apib

```diff
@@ -961,15 +962,572 @@ Outputs the list of peers that were removed.
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Chunked-Output: 1
+        Date: Sat, 06 Feb 2016 22:59:25 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (ndjson)
+        - Extra (string)
+        - ID (Multihash)
+        + Responses (array)
+            + (object)
+                - Addrs (MultiAddr)
+                - ID (Multihash)
+        - Type (number)
+
+    + Body
+
+        ```
+        {
+          "Extra": "",
+          "ID": "QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
+          "Responses": null,
+          "Type": 0
+        }
+        ```
+
+## put [GET /dht/put{?arg1,arg2}{&verbose}]
+Run a `PutValue` query through the DHT
+
+PutValue will store the given key value pair in the dht.
+
+#### Bugs
+
+- While `verbose` is specified in the HTTP API and is used in the HTTP request,
+it returns no extra information in the ndjson response.
+- An invalid or empty key returns the same response as a normal one.
+
++ Parameters
+    + arg1 (string, required) - The key to store the value at.
+    + arg2 (string, required) - The value to store.
+    + verbose (boolean, optional) - Write extra information.
+
++ Request Without Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/dht/put"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/dht/put"
+        ```
+
++ Response 400
+
+    + Headers
+
+        ```
+        Date: Sat, 06 Feb 2016 23:01:30 GMT
+        Content-Length: 26
+        Content-Type: text/plain; charset=utf-8
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        Argument 'key' is required
+        ```
+
++ Request With Empty Or Invalid First Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/dht/put?arg="
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/dht/put?arg="
+        ```
+
++ Response 400
+
+    + Headers
+
+        ```
+        Date: Sat, 06 Feb 2016 23:02:04 GMT
+        Content-Length: 28
+        Content-Type: text/plain; charset=utf-8
+        ```
+
+    + Attributes (string
+
+    + Body
+
+        ```
+        Argument 'value' is required
+        ```
+
++ Request With Arguments
+
+    The response is the same if either argument is empty or invalid. For example:
+
+        curl -i "http://localhost:5001/api/v0/dht/put?arg=&arg="
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/dht/put?arg=kitten"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/dht/put?arg=kitten"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Chunked-Output: 1
+        Date: Sat, 06 Feb 2016 22:59:25 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (ndjson)
+        - Extra (string)
+        - ID (Multihash)
+        + Responses (array)
```

Comment:
Just reran the exact same query and got this: 

- `http://localhost:5001/api/v0/dht/findpeer?arg=QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ`

- https://gist.github.com/dignifiedquire/45dd217fed5ce84bc1f4#file-out-json-L32